spec/lib/whitehall_importer/integrity_checker/unpublishing_check_spec.rb

```diff
@@ -0,0 +1,160 @@
+RSpec.describe WhitehallImporter::IntegrityChecker::UnpublishingCheck do
+  let(:withdrawal_explanation) { "This has been moved" }
+  let(:withdrawal) do
+    build(:withdrawal,
+          public_explanation: "#{withdrawal_explanation} [elsewhere](https://www.gov.uk/elsewhere)")
+  end
+
+  let(:withdrawn_edition) { build(:edition, :withdrawn, withdrawal: withdrawal) }
+
+  let(:removed_explanation) { "This has been removed" }
+  let(:removal) do
+    build(:removal,
+          alternative_url: "/somewhere",
+          explanatory_note: "#{removed_explanation} [Visit here](https://www.gov.uk/here)")
+  end
+
+  let(:removed_edition) { build(:edition, :removed, removal: removal) }
+
+  let(:publishing_api_withdrawal) do
+    {
+      "type" => "withdrawal",
+      "unpublished_at" => withdrawal.withdrawn_at.rfc3339,
+      "explanation" => "#{withdrawal_explanation} <a href=\"https://www.gov.uk/elsewhere\">elsewhere</a>",
+    }
+  end
+
+  let(:publishing_api_gone) do
+    {
+      "type" => "gone",
+      "unpublished_at" => Date.yesterday.end_of_day,
+    }
+  end
+
+  let(:publishing_api_redirect) do
+    {
+      "type" => "redirect",
+      "alternative_path" => "/somewhere",
+      "explanation" => "#{removed_explanation} <a href=\"https://www.gov.uk/here\">Visit here</a>",
+    }
+  end
```

Comment:
I feel like we might be setting up too many variables here that serve almost all test cases - they can often seem to form a smell when one of the tests may need to reach outside them. 

I wonder if it might be nicer doing more of the initialisation of variables inside tests/their relevant context blocks and maybe setting up functions for `publishing_api_redirect` etc.

For example:

```
def publishing_api_withdrawal(attributes = {})
  {
      type: withdrawal,
      unpublished_at: Time.zone.now.rfc3339,
      explanation: "Explanation",
  }.merge(attributes).as_json 
end
```

then in the first test you could probably get away with:

```
      it "returns true if it has a 'withdrawal' unpublishing type in Publishing API" do
        unpublishing_check = described_class.new(build(:edition, :withdrawn),
                                                 publishing_api_withdrawal)
        expect(unpublishing_check.expected_type?).to be true
      end
```

I think this suggestion may end up increasing number of lines a bit but I think the result will be simpler to follow and the tests more focused.

Reply:
I've moved the Publishing API payloads to their own methods.